test/integration/change-streams/change_stream.test.ts

```diff
@@ -2138,6 +2138,185 @@ describe('ChangeStream resumability', function () {
         );
       });
     });
+
+    context('#asyncIterator', function () {
+      for (const { error, code, message } of resumableErrorCodes) {
+        it(
+          `resumes on error code ${code} (${error})`,
+          { requires: { topology: '!single', mongodb: '>=4.2' } },
+          async function () {
+            changeStream = collection.watch([]);
+            await initIteratorMode(changeStream);
+
+            await client.db('admin').command({
+              configureFailPoint: is4_2Server(this.configuration.version)
+                ? 'failCommand'
+                : 'failGetMoreAfterCursorCheckout',
+              mode: { times: 1 },
+              data: {
+                failCommands: ['getMore'],
+                errorCode: code,
+                errmsg: message
+              }
+            } as FailPoint);
+
+            await collection.insertOne({ city: 'New York City' });
+
+            let total_changes = 0;
```

Comment:
I know I recommended this pattern so sorry for changing my mind 😬.  I think we should manually iterate the iterator:

```typescript
const changeStreamIterator = changeStream[Symbol.asyncIterator]()
await changeStreamIterator.next();
```

This also would clean up the code a bit, since we wouldn't need to manually keep track of the count 🙂

Reply:
All good, this is a lot cleaner and easier to maintain anyways.